#!/usr/bin/env python
PACKAGE = 'r2_behavior'

from dynamic_reconfigure.parameter_generator_catkin import *
import yaml_config

gen = ParameterGenerator()
# Tracking parameters
gen.add("enable_flag", bool_t, 0, "Enable animations", True)
gen.add("enable_gestures", bool_t, 0, "Enable Gestures", True)
gen.add("gesture_magnitude", double_t, 0, "Magnitude multiplier", 1, 0, 2)
gen.add("gesture_speed", double_t, 0, "Speed multiplier", 1, 0, 2)
gen.add("time_between_gestures_min", double_t, 0, "Min interval between gestures", 1, 0.0, 15)
gen.add("time_between_gestures_max", double_t, 0, "Max interval between gestures", 5, 0.5, 15)
gen.add("trigger_gesture_on_update", bool_t, 0, "Trigger Gesture on update", False)
gen.add("stop_running_gesture_on_update", bool_t, 0, "Rampout running gesture on update", False)

gen.add("gestures", str_t, 0, 'Gestures', '{}')
# Expressions
gen.add("enable_expressions", bool_t, 0, "Enable Expressions", True)
gen.add("expression_magnitude", double_t, 0, "Expression multiplier", 1, 0, 2)
gen.add("expression_duration", double_t, 0, "Duration multiplier", 1, 0, 2)
gen.add("time_between_expressions_min", double_t, 0, "Min interval between expressions", 5, 1, 15)
gen.add("time_between_expressions_max", double_t, 0, "Max interval between expressions", 8, 1, 15)
gen.add("expressions", str_t, 0, "Expressions", '{}')
# UI schema for gestures and expressions
gen.add("node_schema", str_t, 0, '', yaml_config.parse('animations_schema') or '{}')

# package name, node name, config name
exit(gen.generate(PACKAGE, "animation", "Animation"))
